### PR TITLE
[boschshc] Support Bosch Smart Home Cameras (#12666)

### LINKED ISSUE #12666: [boschshc] Support for Bosch Security Cameras
Hi, I noticed that the Bosch security cameras (360° indoor camera and Eyes outdoor camera) are not supported by the Bosch Smart Home binding yet.

According to https://apidocs.bosch-smarthome.com/local/, both cameras offer only one service (`PrivacyMode`), which can be enabled or disabled. Not much, but better than nothing ;)

```
GET /devices/{deviceId}/services/PrivacyMode/state Retrieve the state of the PrivacyMode service.
PUT /devices/{deviceId}/services/PrivacyMode/state Executes the privacyModeState on the device.
```

Is someone already working on this? Otherwise I would consider contributing.

### PR DESCRIPTION
**Pull request by david-pace**
- add thing definitions for indoor security camera 360 and outdoor
security camera Eyes
- implement services for privacy mode and camera notifications
- implement camera handler
- add unit tests
- update documentation

closes #12666

Signed-off-by: David Pace <dev@davidpace.de>

### CODE REVIEW

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/camera/CameraHandler.java
Comment (david-pace) [inline bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/camera/CameraHandler.java:80]: The long poll results do not include the initial camera states. Is there a more elegent way to obtain the initial states?

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/services/privacymode/PrivacyModeState.java
Comment (david-pace) [inline bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/services/privacymode/PrivacyModeState.java:35]: The API exposes the so called _privacy mode_, which is the opposite of "camera enabled". With the current implementation we expose the API 1:1, meaning if the user switches the privacy mode off, the camera will be on. This might be counterintuitive for the uses, who probably expect that the camera is enabled when the switch is on.

Do you think we should rather offer an inverse logic channel/state? How would you call this channel? Maybe `camera-state`?

**Comment by coeing** [comment]
@david-pace Cool, thanks a lot for your work! 👍 @GerdZanker is the main developer on the binding now, but I will have a quick look at your changes and send some comments if I see something odd :)

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/services/privacymode/PrivacyModeState.java
Comment (coeing) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/services/privacymode/PrivacyModeState.java:35]: I would stay as close to the official API as possible, so keeping the privacy mode here. It may include more than the camera deactivation in the future, maybe it already does. You could add a read-only `camera-state` channel of course which inverts this flag. But writing it should only be possible via the `privacy-mode` channel.

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/camera/CameraHandler.java
Comment (coeing) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/camera/CameraHandler.java:80]: If this becomes more common, we could think about moving the code from your `requestInitial...` methods into the `registerService` method, e.g. with a flag that the initial state should be actively fetched. But if the long poll does not contain the information I do not see a way to get the state otherwise.

**Review by coeing — APPROVED** [review]
Very good code quality! 👍 Hope to see more contributions from you for this binding 🙂 

If you have any feedback on the general code architecture of the binding, let us know!

**Comment by david-pace** [comment]
> Hope to see more contributions from you for this binding 🙂

I'm happy to contribute more code in the future, thank you very much for your feedback 🙂

> If you have any feedback on the general code architecture of the binding, let us know!

So far I found everything designed and thought out very well 👍 But I'll let you know if I have any ideas.

**Review by jlaur — APPROVED** [review]
Looks really good, I have nothing to add. Thanks! If you are ready to merge, please resolve the two comments and/or confirm.

**Comment by david-pace** [comment]
Thank you @jlaur 👍 I would propose waiting for feedback on the two code comments by at least one of @GerdZanker and/or @stefan-kaestle, if that's fine with you.

**Comment by GerdZanker** [comment]
Hello @david-pace and thank you for contributing to the BSH binding.

Your pull request arrived right in time to check my new development environment. I have to say that was block since a too long time because of my ten year old computer. Sorry for this, especially to @coeing. Now I replaced everything inside my mini-tower and the new hardware works perfectly and very fast. Same as your code - I was able to test your new eyes camera handler successfully with my hardware.

So please just check the logging statements to finish the pr.

**Review by GerdZanker — CHANGES_REQUESTED** [review]

**Comment by stefan-kaestle** [comment]
Awesome stuff, thanks @david-pace!

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/camera/CameraHandler.java
Comment (david-pace) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/camera/CameraHandler.java:80]: ok, then I will leave it as it is for now and it can be generalized later if necessary.

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/services/privacymode/PrivacyModeState.java
Comment (david-pace) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/services/privacymode/PrivacyModeState.java:35]: I decided to go only for the `privacy-mode` channel in the initial version that exposes the API 1:1. If required, other channels can be added later.

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/camera/CameraHandler.java
Comment (david-pace) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/camera/CameraHandler.java]: Thanks for calling my attention to the logging guidelines. In this case this would indeed not be a bug and the log level `error` is not applicable here. I changed the log level for all logger calls in those methods to `debug` now. I am not sure whether a stack trace would be allowed or even useful on debug level. In my current commit I have omitted the stack trace. If you are fine with that you can resolve this comment.

**Comment by david-pace** [comment]
Thank you very much for your feedback, @GerdZanker and @stefan-kaestle 😃 I pushed a commit that adresses the logging enhancements 👍

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/camera/CameraHandler.java
Comment (jlaur) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/camera/CameraHandler.java]: It's really fine without stack trace, especially for debug level. Nice catch about error level, @GerdZanker, I somehow missed that.

**Pull request merged**
